Modules/xASL_module_ASL.m

```diff
@@ -520,46 +652,45 @@
     % If BASIL quantification will be performed, only native space analysis is possible
     if isfield(x.Q, 'bUseBasilQuantification') && x.Q.bUseBasilQuantification
         % Quantification in native space only for BASIL
-		if xASL_exist(x.P.Path_PWI4D,'file')
-			% Quantify CBF using PWI4D by default
-			xASL_wrp_Quantify(x, x.P.Path_PWI4D, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient);
-		else
-			% Use PWI if PWI4D does not exist
-            xASL_wrp_Quantify(x, x.P.Path_PWI, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient);
-        end
+
+        xASL_wrp_Quantify(x, x.P.Path_PWI4D, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient);
         % Transform the output volumes of BASIL to standard space (running
         % it in standard space takes too much time and the
         % noise-distributions may differ due to the B-spline interpolation
-        InputPaths = {x.P.Path_CBF, x.P.Path_ATT};
-        OutputPaths = {x.P.Pop_Path_qCBF, x.P.Pop_Path_ATT}; 
+        InputPaths = {x.P.Path_CBF, x.P.Path_ATT, x.P.Path_ABV};
+        OutputPaths = {x.P.Pop_Path_qCBF, x.P.Pop_Path_ATT, x.P.Pop_Path_ABV}; 
         xASL_spm_deformations(x, InputPaths, OutputPaths, 4, [], [], x.P.Path_y_ASL);
     else
         % Quantification in standard space:
         xASL_wrp_Quantify(x);
         % Quantification in native space:
-        xASL_wrp_Quantify(x, x.P.Path_PWI, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient);
+        xASL_wrp_Quantify(x, x.P.Path_PWI4D, x.P.Path_CBF, x.P.Path_rM0, x.P.Path_SliceGradient);
     end
     
-	% allow 4D quantification as well
-	if isfield(x.Q, 'SaveCBF4D') && x.Q.SaveCBF4D==1 && ~x.Q.bUseBasilQuantification
-        nVolumes = size(xASL_io_Nifti2Im(x.P.Path_ASL4D), 4);
-        if nVolumes==1
-            warning('x.Q.SaveCBF4D was requested but only one volume exists, skipping');
-        else
-            fprintf('%s\n','Quantifying ASL timeseries in native space');
-            xASL_wrp_Quantify(x, x.P.Path_PWI4D, x.P.Path_qCBF4D, x.P.Path_rM0, x.P.Path_SliceGradient);
-
-            fprintf('%s\n','Quantifying ASL timeseries in standard space');
-            xASL_wrp_Quantify(x, x.P.Pop_Path_PWI4D, x.P.Pop_Path_qCBF4D);
+	% allow 4D quantification as well, storing CBF4D. This is currently not implemented for multi-PLD/multi-TE (BASIL/FABBER)
+	if isfield(x.Q, 'SaveCBF4D') && x.Q.SaveCBF4D
+        if x.modules.asl.bMultiPLD || x.modules.asl.bMultiTE
+            warning('Saving CBF4D was requested but not implemented yet for multi-PLD or multi-TE, skipping');
+        elseif isfield(x.Q, 'SaveCBF4D') && x.Q.SaveCBF4D && ~(x.modules.asl.bMultiPLD || x.modules.asl.bMultiTE)
```

Comment:
replace by `else`
Because the left part is true because of line 671. And the right part is also false based on line 672. So this condition here is always true.